packages/test/src/test-testenvironment.ts

```diff
@@ -0,0 +1,144 @@
+import anyTest, { TestInterface } from 'ava';
+import { v4 as uuid4 } from 'uuid';
+import { TestWorkflowEnvironment, workflowInterceptorModules } from '@temporalio/testing';
+import { Worker } from '@temporalio/worker';
+import { WorkflowFailedError } from '@temporalio/client';
+import { assertFromWorkflow, sleep, raceActivityAndTimer } from './workflows/testenv-test-workflows';
+
+interface Context {
+  testEnv: TestWorkflowEnvironment;
+}
+
+async function withWorker<R>(worker: Worker, fn: () => Promise<R>): Promise<R> {
+  const runAndShutdown = async () => {
+    try {
+      return await fn();
+    } finally {
+      worker.shutdown();
+    }
+  };
+  const [_, ret] = await Promise.all([worker.run(), runAndShutdown()]);
+  return ret;
+}
+
+const test = anyTest as TestInterface<Context>;
+
+test.before(async (t) => {
+  t.context = {
+    testEnv: await TestWorkflowEnvironment.create({
+      testServer: {
+        stdio: 'inherit',
+      },
+    }),
+  };
+});
+
+test.after.always(async (t) => {
+  await t.context.testEnv?.teardown();
+});
+
+test.serial('TestEnvironment sets up test server and is able to run a Workflow with time skipping', async (t) => {
+  const { workflowClient, nativeConnection } = t.context.testEnv;
+  const worker = await Worker.create({
+    connection: nativeConnection,
+    taskQueue: 'test',
+    workflowsPath: require.resolve('./workflows/testenv-test-workflows'),
+  });
+  await withWorker(worker, async () => {
+    await workflowClient.execute(sleep, {
+      workflowId: uuid4(),
+      taskQueue: 'test',
+      args: [1_000_000],
+    });
+  });
+  t.pass();
+});
+
+test.serial('TestEnvironment can toggle between normal and skipped time', async (t) => {
+  const { workflowClient, nativeConnection } = t.context.testEnv;
+
+  const worker = await Worker.create({
+    connection: nativeConnection,
+    taskQueue: 'test',
+    workflowsPath: require.resolve('./workflows/testenv-test-workflows'),
+  });
+
+  const race = (runInNormalTime: boolean) =>
+    Promise.race([
+      workflowClient
+        .execute(sleep, {
+          workflowId: uuid4(),
+          taskQueue: 'test',
+          args: [1_000_000],
+          runInNormalTime,
+        })
+        .then(() => 'fake time'),
+      // Hopefully 5 seconds will be enough in CI
+      new Promise((resolve) => setTimeout(resolve, 5000)).then(() => 'real time'),
```

Comment:
Heh, you'd be on me about this one :wink: 

I think it is adequate to set a lower value for the sleep timer, and verify that in the skipping case it took less time than it should've, and in the real case it took at least as long as it should've.

Reply:
I'll change to verify it slept for less than a minute, if that's not enough in CI I don't know what is.
I won't actually sleep though.

Reply:
I changed it but it's still racy, IMHO this version is more readable.
Not sure what the best way around it is, if you're testing actual time you just need to wait and the longer you're willing to wait the less flakey the test will be.